torch/_inductor/pattern_matcher.py

```diff
@@ -229,6 +236,14 @@ def find_anchor_nodes(self, ctx: MatchContext, searched):
         if self in ctx.pattern_to_node:
             yield ctx.pattern_to_node[self]
 
+    def pattern_eq(self, other: Any) -> bool:
```

Comment:
Is there a reason to define `pattern_eq` instead of `__eq__` ?

Reply:
I originally defined `__eq__`.  The main reason I changed it is I didn't really want to have to implement `__hash__` since I only care about this for debugging.